cmake/modules/MinGWCrossCompile.cmake

```diff
@@ -16,26 +16,40 @@ SET(CMAKE_RC_COMPILER		${MINGW_TOOL_PREFIX}gcc)
 
 IF(WIN64)
 # specify the cross compiler
-SET(MINGW_TOOL_PREFIX32		${MINGW_PREFIX32}/bin/${CMAKE_SYSTEM_PROCESSOR32}-w64-mingw32-)
-SET(CMAKE_C_COMPILER32		${MINGW_TOOL_PREFIX32}gcc)
-SET(CMAKE_CXX_COMPILER32	${MINGW_TOOL_PREFIX32}g++)
+	SET(MINGW_TOOL_PREFIX32 ${MINGW_PREFIX32}/bin/${CMAKE_SYSTEM_PROCESSOR32}-w64-mingw32-)
+	SET(CMAKE_C_COMPILER32		${MINGW_TOOL_PREFIX32}gcc)
+	SET(CMAKE_CXX_COMPILER32	${MINGW_TOOL_PREFIX32}g++)
 ENDIF()
 
 # specify location of some tools
 SET(STRIP					${MINGW_TOOL_PREFIX}strip)
 SET(WINDRES					${MINGW_TOOL_PREFIX}windres)
-SET(PKG_CONFIG_EXECUTABLE	${MINGW_TOOL_PREFIX}pkg-config)
-SET(PKG_CONFIG_FOUND		TRUE)
+IF(MSYS)
+	SET(ENV{PKG_CONFIG_PATH} "${PKG_CONFIG_PATH}:${MINGW_PREFIX_ALT}/lib/pkgconfig")
+ELSE()	
+	SET(PKG_CONFIG_EXECUTABLE	${MINGW_TOOL_PREFIX}pkg-config)
+	SET(PKG_CONFIG_FOUND		TRUE)
+ENDIF()
 
-SET(QT_BINARY_DIR			${MINGW_PREFIX}/bin)
+SET(QT_BINARY_DIR		${MINGW_PREFIX}/bin)
 SET(QT_QMAKE_EXECUTABLE		${QT_BINARY_DIR}/qmake)
 
 # search for programs in the build host directories
 SET(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
 # for libraries and headers in the target directories
-SET(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
-SET(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+SET(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY BOTH)
+SET(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE BOTH)
+
+IF(MSYS)
+	INCLUDE_DIRECTORIES(${MINGW_PREFIX_ALT}/include)
+	SET(QT_INCLUDE_DIR ${MINGW_PREFIX_ALT}/include/qt4)
```

Comment:
This was an attempt to override the `qtconfig.h` for enabling `QSharedMemory`, which I simply can't figure out how to override properly.  I may need to put our own qt4 higher in order so that `pkg-config` finds it first.